### PR TITLE
Remove context statics stuff

### PR DESCRIPTION
**Pull request by janvorli**
Remove everything related to context statics from the runtime and debugger stuff.

### CODE REVIEW

**Comment by mikem8361** [comment]
How will this affect using SOS against older versions of the runtime? I hope to port all your recent collectible type changes and this one to the diagnostics repo SOS (and eventually remove the one in coreclr).  That SOS needs to work or at least fail gracefully against 1.x, 2.0.x and 2.1.x runtimes.

File: src/debug/inc/dacdbistructures.h [hunk 1]
```diff
@@ -573,7 +573,6 @@ class MSLAYOUT FieldData
     bool            m_fFldIsStatic;           // true if static field, false if instance field
     bool            m_fFldIsRVA;              // true if static relative to module address
     bool            m_fFldIsTLS;              // true if thread-specific static
-    bool            m_fFldIsContextStatic;    // true if context-specific static
```
Comment (mikem8361) [inline src/debug/inc/dacdbistructures.h]: Is this exposed to SOS somehow? Would this break SOS targeting older runtimes?

Comment (janvorli) [reply src/debug/inc/dacdbistructures.h]: I  believe the FieldData structure is not one of those where the layout needs to be preserved. I think that all such structures are in dacprivate.h.
As for the field itself, we have never supported context statics on CoreCLR, so the `m_fFldIsContextStatic` was always false and so the removal of the field usage basically reflects that.

Comment (mikem8361) [reply src/debug/inc/dacdbistructures.h]: Sounds good.

**Comment by janvorli** [comment]
@dotnet-bot test CentOS7.1 x64 Checked Innerloop Build and Test  please

**Pull request merged**